nautobot/extras/filters.py

```diff
@@ -740,6 +745,55 @@ class Meta:
         fields = ["id", "relationship", "source_type", "source_id", "destination_type", "destination_id"]
 
 
+class RelationshipFilter(django_filters.Filter):
+    """
+    Filter objects by the presence of a CustomFieldValue. The filter's name is used as the CustomField name.
+    """
+
+    def __init__(self, relationship, value_list, queryset, *args, **kwargs):
+        self.relationship = relationship
+        self.value_list = value_list
+        self.qs = queryset
+        super().__init__(*args, **kwargs)
+
+    def filter(self, qs, value):
+        if not value:
+            return super().filter(qs, value)
+        else:
+            values = RelationshipAssociation.objects.filter(
+                destination_id__in=self.value_list, source_type=self.relationship.source_type
+            ).values_list("source_id", flat=True)
+            if len(qs) == len(self.qs):
+                qs &= self.get_method(self.qs)(Q(**{"id__in": values}))
+            else:
+                qs |= self.get_method(self.qs)(Q(**{"id__in": values}))
+            return qs
+
+
+class RelationshipAssociationModelFilterSet(BaseFilterSet):
```

Comment:
Would it be possible to add this mixin into `NautobotFilterSet` rather than requiring every filterset to add it? Especially if it can be made to gracefully do nothing if the filterset is for a model that doesn't support relationships.